src/main/java/com/marklogic/kafka/connect/source/RowBatcherSourceTask.java

```diff
@@ -33,20 +51,79 @@ public String version() {
     @Override
     public final void start(Map<String, String> config) {
         logger.info("Starting");
-        Map<String, Object> parsedConfig = MarkLogicSourceConfig.CONFIG_DEF.parse(config);
-        logger.info("Started");
+        logger.info(String.valueOf(config));
+        this.parsedConfig = MarkLogicSourceConfig.CONFIG_DEF.parse(config);
+        DatabaseClientConfig databaseClientConfig = new DefaultDatabaseClientConfigBuilder().buildDatabaseClientConfig(parsedConfig);
+        this.databaseClient = new DefaultConfiguredDatabaseClientFactory().newDatabaseClient(databaseClientConfig);
+        dataMovementManager = databaseClient.newDataMovementManager();
     }
 
     @Override
     public List<SourceRecord> poll() throws InterruptedException {
         logger.info("poll not yet implemented");
 
+        rowBatcher =  getNewRowBatcher();
+        performPoll();
+
         Thread.sleep(1000);
         return null;
     }
 
+    protected RowBatcher<JsonNode> getNewRowBatcher() {
+        rowBatcher =  dataMovementManager.newRowBatcher(jsonHandle);
+        configureRowBatcher(parsedConfig, rowBatcher);
+        return rowBatcher;
+    }
+
+    protected void performPoll() {
+        dataMovementManager.startJob(rowBatcher);
+        rowBatcher.awaitCompletion();
+        rowBatcher = null;
+    }
+
     @Override
     public void stop() {
-        logger.info("Stopping RowBatcherSourceTask");
+        if (rowBatcher != null) {
+            rowBatcher.awaitCompletion();
```

Comment:
Per the javadocs for the `stop()` method in the superclass, I don't think we want to wait for completion - the user intent is to interrupt anything going on. So I think we just want `stopJob`, which will achieve the desired effect here.

Reply:
I think you can test this as well. You can add a custom listener that sleeps for 500ms or so, and then in the test, immediately call "stop" after calling "poll".

Reply:
I put a new test in. Take a look to see if it is what you meant.